mlrun/runtimes/base.py

```diff
@@ -800,6 +800,14 @@ def clean_build_params(self):
         self.spec.build = {}
         return self
 
+    def requires_build(self) -> bool:
+        build = self.spec.build
+        return (
+            build.commands
+            or build.requirements
+            or (build.source and not build.load_source_on_run)
+        )
+
```

Comment:
🎖️

Reply:
Credit goes to @alonmr for the suggestion 👏